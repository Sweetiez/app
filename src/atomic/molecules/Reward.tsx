import * as React from 'react';
import styled from 'styled-components';

import {RewardModel} from '../../model';
import getIcons from '../../utils/icons';
import colors from '../../assets/colors';
import {useDispatch, useSelector} from 'react-redux';
import {rewardsSelector} from '../../store/selectors/cart';
import {addReward, removeReward} from '../../store/actions/cart';
import {Text as SpecialText} from '../atoms';

interface Props {
  reward: RewardModel;
  disabled?: boolean;
}
const Content = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  flex: 1;
`;

const Container = styled.View<{disabled}>`
  flex-direction: row;
  border-radius: 10px;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.07);
  margin: 5px;
  align-items: center;
  background-color: white;
  opacity: ${({disabled}) => (disabled ? 0.4 : 1)};
`;
const StyledImage = styled.Image`
  height: 100px;
  width: 100px;
  border-top-left-radius: 15px;
  border-bottom-left-radius: 15px;
  margin-right: 10px;
`;
const Space = styled.View`
  width: 4px;
`;

const Icon = styled.TouchableOpacity`
  margin-right: 5px;
`;
const Reward: React.FC<Props> = ({reward, disabled}) => {
  const dispatch = useDispatch();

  const isSelected = useSelector(rewardsSelector).find(
    item => item.id === reward.id,
  );

  const iconColor = isSelected ? colors.red : colors.yellow;

  const manageReward = () => {
    if (isSelected) {
      dispatch(removeReward(reward));
    } else {
      dispatch(addReward(reward));
    }
  };

  return (
    <Container disabled={disabled}>
      <StyledImage source={{uri: reward.productImage}} />
      <Content>
        <SpecialText
          content={reward.name + ' (' + reward.cost + ')'}
          size={22}
        />
        <Space />
        <Icon onPress={() => !disabled && manageReward()} disabled={disabled}>
          {getIcons(isSelected ? 'trash' : 'plus', iconColor, 20)}
        </Icon>
      </Content>
    </Container>
  );
};

export default Reward;
