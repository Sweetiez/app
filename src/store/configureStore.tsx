import {createStore, combineReducers} from 'redux';
import cartReducer from './reducers/cart';
import shopReducer from './reducers/shop';
import userReducer from './reducers/user';
import ordersReducer from './reducers/orders';
import recipesReducer from './reducers/recipes';

const rootReducer = combineReducers({
  cart: cartReducer,
  shop: shopReducer,
  user: userReducer,
  orders: ordersReducer,
  recipes: recipesReducer,
});
const confStore = () => {
  return createStore(rootReducer);
};
export default confStore;
