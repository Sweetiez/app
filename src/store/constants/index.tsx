// cart
export const ADD_ITEM_TO_CART = 'ADD_ITEM_TO_CART';
export const REMOVE_ITEM_FROM_CART = 'REMOVE_ITEM_FROM_CART';
export const CLEAR_CART = 'CLEAR_CART';

// shop
export const UPDATE_SHOP = 'UPDATE_SHOP';
export const GET_PRODUCTS = 'GET_PRODUCTS';
export const PRODUCT_ERROR = 'PRODUCT_ERROR';

// user
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const REGISTER = 'REGISTER';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const REGISTER_OK = 'REGISTER_OK';
export const REGISTER_ERROR = 'REGISTER_ERROR';
export const SET_USER = 'SET_USER';
export const GET_USER_ERROR = 'GET_USER_ERROR';
export const UPDATE_OK = 'UPDATE_OK';
export const UPDATE_ERROR = 'UPDATE_ERROR';
export const UPDATE_PASSWORD_OK = 'UPDATE_PASSWORD_OK';
export const UPDATE_PASSWORD_ERROR = 'UPDATE_PASSWORD_ERROR';

// evaluation
export const COMMENT_ERROR = 'COMMENT_ERROR';
export const COMMENT_OK = 'COMMENT_OK';
export const REPORT_OK = 'REPORT_OK';
export const REPORT_ERROR = 'REPORT_ERROR';
export const REPORT_ALREADY_MADE = 'REPORT_ALREADY_MADE';

// orders
export const GET_ORDERS_ERROR = 'GET_ORDERS_ERROR';
export const SET_ORDERS = 'SET_ORDERS';
export const CREATE_AN_ORDER_ERROR = 'CREATE_AN_ORDER_ERROR';
export const CREATE_PAYMENT_INTENT_ERROR = 'CREATE_AN_ORDER_ERROR';

// recipes
export const UPDATE_RECIPES = 'UPDATE_RECIPES';
export const RECIPES_ERROR = 'RECIPES_ERROR';
export const RECIPE_ERROR = 'RECIPE_ERROR';
