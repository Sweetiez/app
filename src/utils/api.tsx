import Config from 'react-native-config';
import {Platform} from 'react-native';

export const getUrl = (path: string) => {
  if (Platform.OS === 'ios') {
    return Config.API_IOS_LOCAL_URL + path;
  } else {
    return Config.API_URL + path;
  }
};

export const buildHeaders = token => {
  return {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: token ? 'Bearer ' + token : undefined,
  };
};

export const buildRequest = async (
  method,
  url,
  data,
  onError,
  onSuccess,
  token,
) => {
  const headers = buildHeaders(token);
  try {
    const response = await fetch(getUrl(url), {
      method: method,
      headers,
      body: JSON.stringify(data),
    });
    if (response.status === 200 || response.status === 201) {
      return onSuccess(response);
    } else {
      return onError();
    }
  } catch (error) {
    return onError();
  }
};
