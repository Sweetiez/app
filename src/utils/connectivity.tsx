import NetInfo from '@react-native-community/netinfo';

export const checkConnectivity = async () => {
  const status = await NetInfo.fetch();
  return status.isConnected;
};
