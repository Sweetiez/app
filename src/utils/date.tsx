//@ts-ignore
import moment from 'moment';

export const formatDate = (date: string) => {
  return moment(date).format('DD/MM/YYYY');
};
export const formatDashDate = (date: string) => {
  return moment(date).format('YYYY-MM-DD');
};
