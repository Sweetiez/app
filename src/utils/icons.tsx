// @ts-nocheck
import React from 'react';
import Pen from './../assets/icons/pen.svg';
import Minus from './../assets/icons/minus.svg';
import Plus from './../assets/icons/plus.svg';
import Add from './../assets/icons/add.svg';
import Back from './../assets/icons/arrow-back.svg';
import CookBook from './../assets/icons/cook-book.svg';
import Cake from './../assets/icons/cake.svg';
import Event from './../assets/icons/event.svg';
import Account from './../assets/icons/account.svg';
import Eye from './../assets/icons/eye.svg';
import Interrogation from './../assets/icons/interrogation.svg';
import Cart from './../assets/icons/cart.svg';
import Trash from './../assets/icons/trash.svg';
import NoImage from './../assets/icons/no-image.svg';
import Edit from './../assets/icons/edit.svg';
import Warning from './../assets/icons/warning.svg';
import Orders from './../assets/icons/orders.svg';
import Filter from './../assets/icons/filter.svg';
import Cross from './../assets/icons/cross.svg';
import Circle from './../assets/icons/circle.svg';
import colors from '../assets/colors';

const getIcons = (iconName, color = colors.white, size = 18) => {
  const props = {
    width: size,
    height: size,
    color: color,
  };
  switch (iconName) {
    case 'pen':
      return <Pen {...props} />;
    case 'eye':
      return <Eye {...props} />;
    case 'minus':
      return <Minus {...props} />;
    case 'plus':
      return <Plus {...props} />;
    case 'add':
      return <Add {...props} />;
    case 'back':
      return <Back {...props} />;
    case 'cake':
      return <Cake {...props} />;
    case 'event':
      return <Event {...props} />;
    case 'cookBook':
      return <CookBook {...props} />;
    case 'account':
      return <Account {...props} />;
    case 'interrogation':
      return <Interrogation {...props} />;
    case 'cart':
      return <Cart {...props} />;
    case 'trash':
      return <Trash {...props} />;
    case 'noImage':
      return <NoImage {...props} />;
    case 'edit':
      return <Edit {...props} />;
    case 'warning':
      return <Warning {...props} />;
    case 'orders':
      return <Orders {...props} />;
    case 'filter':
      return <Filter {...props} />;
    case 'cross':
      return <Cross {...props} />;
    case 'circle':
      return <Circle {...props} />;
  }
};

export default getIcons;
