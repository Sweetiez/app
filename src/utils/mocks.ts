jest.mock('./../assets/icons/pen.svg', () => 'Pen');
jest.mock('./../assets/icons/star.svg', () => 'Star');
jest.mock('./../assets/icons/plus.svg', () => 'Plus');
jest.mock('./../assets/icons/minus.svg', () => 'Minus');
jest.mock('./../assets/icons/add.svg', () => 'Add');
jest.mock('./../assets/icons/arrow-back.svg', () => 'Back');
jest.mock('./../assets/icons/cook-book.svg', () => 'CookBook');
jest.mock('./../assets/icons/cake.svg', () => 'Cake');
jest.mock('./../assets/icons/event.svg', () => 'Event');
jest.mock('./../assets/icons/eye.svg', () => 'Eye');
jest.mock('./../assets/icons/trash.svg', () => 'Trash');
jest.mock('./../assets/icons/no-image.svg', () => 'NoImage');
jest.mock('./../assets/icons/edit.svg', () => 'Edit');
jest.mock('./../assets/icons/warning.svg', () => 'Warning');

jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}))



jest.mock('react-native-picker-select', () => 'RNPickerSelect')
jest.mock('react-native-config', () => 'Config')