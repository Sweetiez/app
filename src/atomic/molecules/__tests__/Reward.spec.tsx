import * as React from 'react';
import renderer from 'react-test-renderer';
import '../../../utils/mocks';
import Reward from '../Reward';
import {store} from '../../../store/configureStore';
import ReduxProvider from '../../../utils/provider';

describe('Reward', () => {
  const reward = {
    id: '1',
    name: 'Friandise',
    cost: 12,
    productId: '22',
    productType: 'SWEET',
    productName: 'Tartelette',
    productImage: undefined,
    state: 'PUBLISHED',
  };
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ReduxProvider reduxStore={store}>
          <Reward reward={reward} disabled={false} />
        </ReduxProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
