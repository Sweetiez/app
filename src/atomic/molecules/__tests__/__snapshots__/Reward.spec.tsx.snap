// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reward renders correctly 1`] = `
<View
  disabled={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "flexDirection": "row",
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "shadowColor": "rgba(0, 0, 0, 0.07)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 15,
          "borderTopLeftRadius": 15,
          "height": 100,
          "marginRight": 10,
          "width": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      color="#000000"
      size={22}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Pompiere-Regular",
            "fontSize": 22,
            "textDecorationColor": "black",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
      textDecoration="none"
      textTransform="none"
    >
      Friandise (12)
    </Text>
    <View
      style={
        Array [
          Object {
            "width": 4,
          },
        ]
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Plus
        color="#dba970"
        height={20}
        width={20}
      />
    </View>
  </View>
</View>
`;
